SharedPackages/VPN/Sources/VPN/WideEvent/VPNConnectionWideEventData.swift

```diff
@@ -0,0 +1,237 @@
+
+    // Per-step durations
+    public var browserStartDuration: WideEvent.MeasuredInterval?
+    public var controllerStartDuration: WideEvent.MeasuredInterval?
+    public var oauthDuration: WideEvent.MeasuredInterval?
+    public var tunnelStartDuration: WideEvent.MeasuredInterval?
+
+    // Per-step errors
+    public var browserStartError: WideEventErrorData?
+    public var controllerStartError: WideEventErrorData?
+    public var oauthError: WideEventErrorData?
+    public var tunnelStartError: WideEventErrorData?
+
+    public var errorData: WideEventErrorData?
+
+    public init(extensionType: ExtensionType,
+                startupMethod: StartupMethod,
+                overallDuration: WideEvent.MeasuredInterval? = nil,
+                browserStartDuration: WideEvent.MeasuredInterval? = nil,
+                controllerStartDuration: WideEvent.MeasuredInterval? = nil,
+                oauthDuration: WideEvent.MeasuredInterval? = nil,
+                tunnelStartDuration: WideEvent.MeasuredInterval? = nil,
+                browserStartError: WideEventErrorData? = nil,
+                controllerStartError: WideEventErrorData? = nil,
+                oauthError: WideEventErrorData? = nil,
+                tunnelStartError: WideEventErrorData? = nil,
+                errorData: WideEventErrorData? = nil,
+                contextData: WideEventContextData,
+                appData: WideEventAppData = WideEventAppData(),
+                globalData: WideEventGlobalData = WideEventGlobalData()) {
+        self.extensionType = extensionType
+        self.startupMethod = startupMethod
+        self.overallDuration = overallDuration
+
+        // Per-step latencies
+        self.browserStartDuration = browserStartDuration
+        self.controllerStartDuration = controllerStartDuration
+        self.oauthDuration = oauthDuration
+        self.tunnelStartDuration = tunnelStartDuration
+
+        // Per-step errors
+        self.browserStartError = browserStartError
+        self.controllerStartError = controllerStartError
+        self.oauthError = oauthError
+        self.tunnelStartError = tunnelStartError
+
+        self.errorData = errorData
+        self.contextData = contextData
+        self.appData = appData
+        self.globalData = globalData
+    }
+
+    private static let featureName = "vpn-connection"
+}
+
+// MARK: - Public
+
+extension VPNConnectionWideEventData {
+
+    public enum ExtensionType: String, Codable, CaseIterable {
+        case app
+        case system
+    }
+
+    public enum StartupMethod: String, Codable, CaseIterable {
+        case automaticOnDemand = "automatic_on_demand"
+        case manualByMainApp = "manual_by_main_app"
+        case manualByTheSystem = "manual_by_the_system"
+    }
+
+    public enum StatusReason: String {
+        case partialData = "partial_data"
+        case timeout
+    }
+
+    public enum Step: String, Codable, CaseIterable {
+        case browserStart = "browser_start"
```

Comment:
Is it intentional that this is not being used?  I suspect we'll want this for macOS given the name.

I know this is a bit nitpicky, but I'd suggest adding this when it's used so we can evaluate it in the right context.